src/backend/oneapi/solve.cpp

```diff
@@ -59,32 +60,13 @@ Array<T> solveLU(const Array<T> &A, const Array<int> &pivot, const Array<T> &b,
         ::oneapi::mkl::lapack::getrs_scratchpad_size<compute_t<T>>(
             getQueue(), opts, N, NRHS, LDA, LDB);
 
-    // TODO: which one?
-    Array<intl> ipiv              = cast<intl, int>(pivot);
-    sycl::buffer<int64_t> ipivBuf = ipiv.get()->reinterpret<int64_t, 1>();
-    sycl::buffer<compute_t<T>> scratchpad(scratchpad_size);
-
-    /*
-    sycl::buffer<int64_t> ipivBuf(pivot.elements());
-    getQueue()
-        .submit([&](sycl::handler &h) {
-            auto ipivIn =
-                pivot.get()->template get_access<sycl::access_mode::read>(h);
-            auto ipivOut =
-                ipivBuf.get_access<sycl::access_mode::write>(h);
-            h.parallel_for(pivot.elements(),
-                    [=](sycl::id<1> i) {
-                        ipivOut[i] = static_cast<int64_t>(ipivIn[i]);
-                    });
-
-        });
-    */
-
-    Array<compute_t<T>> B = copyArray<compute_t<T>>(b);
-    sycl::buffer<compute_t<T>> aBuf =
-        A.template getBufferWithOffset<compute_t<T>>();
-    sycl::buffer<compute_t<T>> bBuf =
-        B.template getBufferWithOffset<compute_t<T>>();
+    Array<intl> ipiv        = cast<intl, int>(pivot);
+    buffer<int64_t> ipivBuf = ipiv.get()->reinterpret<int64_t, 1>();
```

Comment:
Is it not accepting intl? It sucks that we need pivot to be converted. Can you update the API to accept s64 types as the pivot?